sequencer/src/persistence/sql.rs

```diff
@@ -186,7 +196,8 @@ impl PersistenceOptions for Options {
     type Persistence = Persistence;
 
     async fn create(self) -> anyhow::Result<Persistence> {
-        SqlStorage::connect(self.try_into()?).await
+        let storage = SqlStorage::connect(self.try_into()?).await?;
+        Ok(Arc::new(RwLock::new(storage)))
```

Comment:
Like I think this is gonna result in a nested `RwLock` pair when we `RwLock` it again to share it into the web server state